File: LondonDataServices.IDecide.Core/Models/Foundations/Consumers/Consumer.cs
﻿// ---------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------

using System;
using System.Collections.Generic;
using LondonDataServices.IDecide.Core.Models.Bases;
using LondonDataServices.IDecide.Core.Models.Foundations.ConsumerAdoptions;
using Microsoft.AspNetCore.Mvc.ModelBinding;

namespace LondonDataServices.IDecide.Core.Models.Foundations.Consumers
{
    public class Consumer : IKey, IAudit
    {
        public Guid Id { get; set; }
        public string EntraId { get; set; }
        public string Name { get; set; }
        public string ContactPerson { get; set; }
        public string ContactNumber { get; set; }
        public string ContactEmail { get; set; }
        public string CreatedBy { get; set; }
        public DateTimeOffset CreatedDate { get; set; }
        public string UpdatedBy { get; set; }
        public DateTimeOffset UpdatedDate { get; set; }

        [BindNever]
        public List<ConsumerAdoption> ConsumerAdoptions { get; set; } = new List<ConsumerAdoption>();
    }
}

File: LondonDataServices.IDecide.Manage.Client/src/models/consumers/consumer.ts
import { ConsumerAdoption } from '../consumerAdoptions/consumerAdoption';

export class Consumer {
    public id: string;
    public entraId?: string;
    public name?: string;
    public contactPerson?: string;
    public contactNumber?: string;
    public contactEmail?: string;
    public createdBy?: string;
    public createdDate?: Date;
    public updatedBy?: string;
    public updatedDate?: Date;
    public consumerAdoptions?: ConsumerAdoption[];

    constructor(consumer?: Partial<Consumer>) {
        this.id = consumer?.id || '';
        this.entraId = consumer?.entraId || '';
        this.name = consumer?.name || '';
        this.contactPerson = consumer?.contactPerson || '';
        this.contactNumber = consumer?.contactNumber || '';
        this.contactEmail = consumer?.contactEmail || '';
        this.createdBy = consumer?.createdBy || '';
        this.createdDate = consumer?.createdDate ? new Date(consumer.createdDate) : undefined;
        this.updatedBy = consumer?.updatedBy || '';
        this.updatedDate = consumer?.updatedDate ? new Date(consumer.updatedDate) : undefined;
        this.consumerAdoptions = consumer?.consumerAdoptions || [];
    }
}